### PR TITLE
Fixing intervals

### PR DESCRIPTION
**Pull request by twicki**
## Technical Description

Fixing the behaviour of intervals:
Intervals were left untouched so far but this leads to namespace clashes in the most recent refactoring of code-gen.
This removes the interval-definitions in the code-generated files

### CODE REVIEW

File: scripts/travis/gtclang-driver-test.sh [hunk 1]
```diff
@@ -49,11 +49,13 @@ cmake .. -DCMAKE_CXX_COMPILER="$CXX"
          -DGTCLANG_UNIT_TESTING=ON                                                                                            \
          -DGTCLANG_INTEGRATION_TESTING=ON                                                                                     \
          -DDawn_DIR=${build_dir}/dawn-prefix/src/dawn-build/prefix/dawn/cmake/                                                \
-         -DPYTHON_EXECUTABLE="$PYTHON_DIR"                                                                                    \
+         -DPYTHON_EXECUTABLE="$PYTHON_DIR"                                                                        			  \
       || fatal_error "failed to configure"
 
 make -j2 || fatal_error "failed to build"
 
+bash ${build_dir}/gtclang-prefix/src/gtclang-build/gtclang-tester-no-codegen.sh
```
Comment (cosunae) [inline scripts/travis/gtclang-driver-test.sh]: why adding this? isnt part of the ctest?

Comment (twicki) [reply scripts/travis/gtclang-driver-test.sh]: this should not be here... I don't know why it is

File: src/gtclang/Frontend/GTClangASTVisitor.cpp [hunk 2]
```diff
@@ -27,6 +27,25 @@ GTClangASTVisitor::GTClangASTVisitor(GTClangContext* context)
...
+bool GTClangASTVisitor::VisitDecl(clang::Decl* Decl) {
+  const auto& SM = context_->getSourceManager();
+  // Only parse declaration of the main-file
+  if(SM.getFileID(Decl->getLocation()) == SM.getMainFileID()) {
+    // We need it to be a var-decl
+    if(llvm::isa<clang::VarDecl>(Decl)) {
+      clang::VarDecl* vd = llvm::dyn_cast<clang::VarDecl>(Decl);
+      if(vd->getType().getBaseTypeIdentifier()) {
+          // we check for interval-defn's that are not the autogenerated loopvariables
+        if(vd->getType().getBaseTypeIdentifier()->getName() == "interval" && vd->getName()!="__k_loopvar__") {
```
Comment (cosunae) [inline src/gtclang/Frontend/GTClangASTVisitor.cpp]: what is this k_loopvar?

Comment (twicki) [reply src/gtclang/Frontend/GTClangASTVisitor.cpp]: the prerprocessor replaces instances of 
```
vertical_region(k_begin, k_finish)
```
with 
```
for(auto __k_loopvar__ : {k_begin, k_finish}
```
and therefore it has the same type and would have been removed.

**Comment by cosunae** [comment]
launch jenkins

**Pull request merged**